tests/unit/test_jobs.py

```diff
@@ -0,0 +1,33 @@
+from catkin_tools.jobs.utils import merge_envs
+
+
+def test_merge_envs_basic():
+    job_env = { 'PATH': '/usr/local/bin:/usr/bin', 'FOO': 'foo' }
+
+    merge_envs(job_env, [
+        { 'PATH': '/usr/local/bin:/bar/baz/bin' },
+        { 'BAR': 'bar' } ])
+
+    # Validate that the known path was not moved from the existing order, and the unfamiliar
+    # path was correctly prepended.
+    assert job_env['PATH'] == '/bar/baz/bin:/usr/local/bin:/usr/bin'
+
+    # Confirm that a key only in the original env persists.
+    assert job_env['FOO'] == 'foo'
```

Comment:
It would be good to see a non-PATH env var that contains a `:` as a sanity check, especially since `ROS_MASTER_URI` tripped it up in the beginning if I remember correctly.